pkg/agent/cluster.go

```diff
@@ -70,9 +71,34 @@ func NewCluster(ctx context.Context, assetDir, rendezvousIP, kubeconfigPath, ssh
 	czero := &Cluster{}
 	capi := &clientSet{}
 
-	authToken, err := FindAuthTokenFromAssetStore(assetDir)
-	if err != nil {
-		logrus.Fatal(err)
+	var authToken string
+	var err error
+
+	authToken, _ = FindAuthTokenFromAssetStore(assetDir) //nolint:errcheck
+
+	switch workflowType {
+	case workflow.AgentWorkflowTypeInstall:
+		// do nothing. The auth tokens do not expire.
+	case workflow.AgentWorkflowTypeAddNodes:
+		// authToken was not found in the asset store, retrieve it from cluster
+		if authToken == "" && kubeconfigPath != "" {
+			authToken, _ = gencrypto.GetAuthTokenFromCluster(ctx, kubeconfigPath)
+			if authToken == "" {
+				logrus.Fatal("Auth token secret not found in the cluster.")
+			}
+		}
+		// check if authToken is expired
```

Comment:
Should the monitor-add-nodes check the final expiration of 48 hours or first 24 hours like add-nodes command to create nodes ISO?